lib/Parse/ParseStmt.cpp

```diff
@@ -228,8 +229,16 @@ Parser::ParseStatementOrDeclarationAfterAttributes(StmtVector &Stmts,
   case tok::kw_default:             // C99 6.8.1: labeled-statement
     return ParseDefaultStatement();
 
+  case tok::kw__Checked:
+    // Checked C - check if checked scope keyword
+    assert(NextToken().is(tok::l_brace));
```

Comment:
In checked c specification, checked keyword can be used as
checked array, checked function, checked scope (compound or struct/union)

Since checked array & checked function is used in declaration, 
checked scope keyword is parsed in declaration
In this case, checked scope keyword is used before l_brace and function 

function (ParseStatementOrDeclarationAfterAttributes) is used to parse statement
so, I have expected that checked keyword is used only before l_brace
In that case, we prepare to parse next l_brace by calling parse compound statement
Otherwise, it is an error case

I agree your opinion. It is not always correct
To handle error case, I will change it